crates/parser/src/config/csv.rs

```diff
@@ -1,227 +0,0 @@
-        })).unwrap()
-    }
-}
-
-impl<'a> TryFrom<&'a str> for LineEnding {
-    type Error = &'static str;
-
-    fn try_from(value: &'a str) -> Result<Self, Self::Error> {
-        if value == "\r\n" {
-            Ok(LineEnding::CRLF)
-        } else if value.len() == 1 {
-            Char::try_from(value).map(LineEnding::Other)
-        } else {
-            Err(r#"expected either a single character or the string "\r\n""#)
-        }
-    }
-}
-
-impl<'de> de::Deserialize<'de> for LineEnding {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: serde::Deserializer<'de>,
-    {
-        let s = std::borrow::Cow::<'de, str>::deserialize(deserializer)?;
-        LineEnding::try_from(s.as_ref()).map_err(|e| de::Error::custom(e))
-    }
-}
-
-impl ser::Serialize for LineEnding {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: serde::Serializer,
-    {
-        match self {
-            LineEnding::CRLF => serializer.serialize_str("\r\n"),
-            LineEnding::Other(c) => c.serialize(serializer),
-        }
-    }
-}
-
-#[derive(Debug, Clone, PartialEq, Default, JsonSchema, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub struct CharacterSeparatedConfig {
-    /// Manually specified headers, which can be used in cases where the file itself doesn't
-    /// contain a header row. If specified, then the parser will assume that the first row is data,
-    /// not column names, and the column names given here will be used. The column names will be
-    /// matched with the columns in the file by the order in which they appear here.
-    #[serde(default)]
-    pub headers: Vec<String>,
-    /// The delimiter that separates values within each row. Only single-byte delimiters are
-    /// supported.
-    #[serde(default)]
-    pub delimiter: Option<Char>,
-    #[serde(default)]
-    /// The value that terminates a line. Only single-byte values are supported, withe the
-    /// exception of "\r\n" (CRLF), which will accept lines terminated by
-    /// _either_ a carriage return, a newline, or both.
-    pub line_ending: Option<LineEnding>,
-    /// The character used to quote fields.
-    #[serde(default)]
-    pub quote: Option<Char>,
-    /// The escape character, used to escape quotes within fields.
-    #[serde(default)]
-    pub escape: Option<Char>,
-    /// The character encoding of the source file. If unspecified, then the parser will make a
-    /// best-effort guess based on peeking at a small portion of the beginning of the file. If
-    /// known, it is best to specify. Encodings are specified by their WHATWG label.
-    #[serde(default)]
-    pub encoding: Option<EncodingRef>,
-    /// Allows a percentage of errors to be ignored without failing the entire
-    /// parsing process. When this limit is exceeded, parsing halts.
-    #[serde(default)]
-    pub error_threshold: Option<ErrorThreshold>,
-}
-
-impl CharacterSeparatedConfig {
```

Comment:
Was this file already not used?

Reply:
yeah, it had been moved to `character_separated.rs` ages ago, and somehow the old one had stuck around. There was no `mod` declaration for it, so it definitely wasn't being used by anything.